### PR TITLE
feat: add support for encode&decode

### PR DESCRIPTION
**Pull request by BanBuDu0**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

According to https://github.com/risingwavelabs/risingwave/issues/8937.
The encode and decode function are implemented in this pull request.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [x] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by lmatz** [comment]
Could you help enable the test cases here: https://github.com/risingwavelabs/risingwave/blob/0bdc988be5cb8b307e204815f9237c6949f664c9/src/tests/regress/data/sql/strings.sql#L649

**Comment by BanBuDu0** [comment]
> `SELECT encode(('\x' || repeat('1234567890abcdef0001', 7))::bytea, 'base64');` `SELECT decode(encode(('\x' || repeat('1234567890abcdef0001', 7))::bytea, 'base64'), 'base64');` These two sqls will produce an error: `ERROR: QueryError: Bind error: cannot cast type "varchar" to "bytea" in Explicit context`
> 
> Seems like an error happend in compile-time which cast expr bind error. I see that no Bytea type mentioned in `CAST_MAP` https://github.com/risingwavelabs/risingwave/blob/main/src/expr/src/sig/cast.rs#L51

I see that the cast function [varchar->bytea] was already implemented here https://github.com/risingwavelabs/risingwave/blob/main/src/expr/src/vector_op/cast.rs#L196

So I fix it by simple add Bytea type to CAST_MAP, see commit：https://github.com/risingwavelabs/risingwave/pull/9351/commits/432b7052926f65eebf2f17ef420ee03095dac9b4
Is there something wrong with this fix?

File: proto/expr.proto [hunk 1]
```diff
@@ -117,6 +117,8 @@ message ExprNode {
     SQRT = 256;
     DEGREES = 257;
     RADIANS = 258;
+    ENCODE = 259;
+    DECODE = 260;
```
Comment (xiangjinwu) [inline proto/expr.proto:121]: Historically we have mixed both number-related and string-related expressions to 2xx. Could you put these 2 new string/bytea expressions starting 310 with a comment saying "string/bytea expressions"?

File: src/tests/regress/data/sql/strings.sql [hunk 2]
```diff
@@ -648,13 +648,12 @@ select md5('12345678901234567890123456789012345678901234567890123456789012345678
 --
 -- encode/decode
 --
---@ SELECT encode('\x1234567890abcdef00', 'hex');
---@ SELECT decode('1234567890abcdef00', 'hex');
---@ SELECT encode(('\x' || repeat('1234567890abcdef0001', 7))::bytea, 'base64');
---@ SELECT decode(encode(('\x' || repeat('1234567890abcdef0001', 7))::bytea,
---@                      'base64'), 'base64');
```
Comment (xiangjinwu) [inline src/tests/regress/data/sql/strings.sql:655]: Keep them on 2 lines with the leading indentation spaces. In other words, only remove the 4 leading chars of `--@ ` and avoid any other changes, or the test output would not match.

**Review by xiangjinwu — COMMENTED** [review]
Thanks for your contribution. And your fix to the `CAST_MAP` is correct.

**Comment by BanBuDu0** [comment]
After I changed the CAST_MAP, I should also modify the e2e_test/batch/catalog/pg_cast.slt.part following the change of cast_map.
By the way, I also fixed the from_oid function here https://github.com/risingwavelabs/risingwave/blob/main/src/common/src/types/postgres_type.rs#L48, there was no Bytea before.
See https://github.com/risingwavelabs/risingwave/pull/9351/commits/186bc936037e940fae7fc79ebf2710c2b19022ce

Finally, all the test was passsed now🥳.

**Review by lmatz — APPROVED** [review]
Many thanks for your contribution!

**Pull request merged**